File: python/lld/cache-design/key-value-store/entry.py
from dataclasses import dataclass
from typing import Generic, TypeVar

V = TypeVar("V")


@dataclass
class Entry(Generic[V]):
    """An immutable-ish snapshot of one stored value: what it holds, when
    it expires, and a version counter for optimistic-concurrency writes."""
    value: V
    expires_at: float
    version: int

File: java/lld/cache-design/key-value-store/Entry.java
// An immutable-ish snapshot of one stored value: what it holds, when it
// expires, and a version counter for optimistic-concurrency writes.
public class Entry<V> {
    public final V value;
    public final long expiresAt;
    public final int version;

    public Entry(V value, long expiresAt, int version) {
        this.value = value; this.expiresAt = expiresAt; this.version = version;
    }
}